### PR TITLE
Move session persistence package to legacy-private module

### LINKED ISSUE #12712: Move session persistence package to legacy-private module
## Description

The (non-)persistence of sessions should not be visible to services.

Move [`org.keycloak.models.session`](https://github.com/keycloak/keycloak/tree/main/server-spi-private/src/main/java/org/keycloak/models/session) package to the legacy module.

### PR DESCRIPTION
**Pull request by ahus1**
Closes #12712

### CODE REVIEW

**Comment by ahus1** [comment]
No public APIs have been changed, therefore no docs PR for this one. Test are green!

File: testsuite/integration-arquillian/tests/base/pom.xml [hunk 1]
```diff
@@ -852,6 +852,7 @@
                                 <keycloak.authorizationCache.enabled>false</keycloak.authorizationCache.enabled>
                                 <keycloak.realmCache.enabled>false</keycloak.realmCache.enabled>
                                 <keycloak.userCache.enabled>false</keycloak.userCache.enabled>
+                                <keycloak.userSessionPersister.provider></keycloak.userSessionPersister.provider>
```
Comment (mhajas) [inline testsuite/integration-arquillian/tests/base/pom.xml:855]: Why is this needed?

**Review by mhajas — COMMENTED** [review]
Thanks @ahus1, the PR looks good I just added one question

File: testsuite/integration-arquillian/tests/base/pom.xml (hunk 1 above)
Comment (mhajas) [reply testsuite/integration-arquillian/tests/base/pom.xml:855]: I would just expect this is not necessary as `userSessionPersister` is disabled by the `isSupported` method. Should this be done also for?:
https://github.com/keycloak/keycloak/blob/7d96f3ad5ad32c4a78a9d571ad891ae953258332/testsuite/integration-arquillian/tests/base/src/test/resources/META-INF/keycloak-server.json#L180-L182

Comment (mhajas) [reply testsuite/integration-arquillian/tests/base/pom.xml:855]: We decided to investigate `userFederatedStorage` in a separate issue https://github.com/keycloak/keycloak/issues/12895

**Review by mhajas — APPROVED** [review]
Thanks @ahus1, approving
